### PR TITLE
Add capabilities methods to `ExtendedGameTestHelper`

### LINKED ISSUE #463: getCapability fails because relative coordinates are used in the tests
**Minecraft Version:** 1.20.4

**NeoForge Version:** 20.4.70-beta

**Description of issue:**
If I use the Minecraft in-game testing framework, the method 
```java
default <T, C> @Nullable T getCapability(BlockCapability<T, C> cap, BlockPos pos, C context
```
does not take into account the relative coordinates that are used.
So for now the only way to solve this problem is to retrieve the BE and then pass it to the method.

### PR DESCRIPTION
**Pull request by Sirttas**
This PR add a few methods to get capabilities using the `ExtendedGameTestHelper` from the test framework.

### CODE REVIEW

File: testframework/src/main/java/net/neoforged/testframework/gametest/ExtendedGameTestHelper.java [hunk 1]
```diff
@@ -188,6 +191,33 @@ public <T extends BlockEntity> T requireBlockEntity(int x, int y, int z, Class<T
...
     }
 
+
+    @Nullable
+    public <T, C> T getCapability(BlockCapability<T, C> cap, BlockPos pos, C context) {
+        return getLevel().getCapability(cap, absolutePos(pos), context);
+    }
+
+    @Nullable
+    public <T, C> T getCapability(BlockCapability<T, C> cap, BlockPos pos, @Nullable BlockState state, @Nullable BlockEntity blockEntity, C context) {
```
Comment (Technici4n) [inline testframework/src/main/java/net/neoforged/testframework/gametest/ExtendedGameTestHelper.java]: This overload is probably not necessary, it only exists for performance reasons and I don't think that this function is perf-sensitive. :)

File: testframework/src/main/java/net/neoforged/testframework/gametest/ExtendedGameTestHelper.java [hunk 2]
```diff
@@ -188,6 +191,33 @@ public <T extends BlockEntity> T requireBlockEntity(int x, int y, int z, Class<T
...
+
+    public <T, C> T requireCapability(BlockCapability<T, C> cap, BlockPos pos, C context) {
+        final var capability = getCapability(cap, pos, context);
+        if (capability == null) {
+            throw new GameTestAssertPosException("Expected capability " + cap + " but there was none", absolutePos(pos), pos, getTick());
+        }
+        return capability;
+    }
+
+    public <T, C> T requireCapability(BlockCapability<T, C> cap, BlockPos pos, @Nullable BlockState state, @Nullable BlockEntity blockEntity, C context) {
```
Comment (Technici4n) [inline testframework/src/main/java/net/neoforged/testframework/gametest/ExtendedGameTestHelper.java]: Same comment here

**Review by Technici4n — CHANGES_REQUESTED** [review]

File: testframework/src/main/java/net/neoforged/testframework/gametest/ExtendedGameTestHelper.java (hunk 1 above)
Comment (Sirttas) [reply testframework/src/main/java/net/neoforged/testframework/gametest/ExtendedGameTestHelper.java]: Agree.

Comment (Sirttas) [reply testframework/src/main/java/net/neoforged/testframework/gametest/ExtendedGameTestHelper.java]: Done.

File: testframework/src/main/java/net/neoforged/testframework/gametest/ExtendedGameTestHelper.java (hunk 2 above)
Comment (Sirttas) [reply testframework/src/main/java/net/neoforged/testframework/gametest/ExtendedGameTestHelper.java]: Done.

**Review by Technici4n — APPROVED** [review]

**Pull request merged**